src/xml2js.coffee

```diff
@@ -255,6 +257,8 @@ class exports.Parser extends events.EventEmitter
       obj = {}
       obj[charkey] = ""
       unless @options.ignoreAttrs
+        if @options.explicitAttrs and attrkey not of obj
```

Comment:
Wouldn't it make sense to put the test at the end? Especially since line 264 does exactly the same?